src/engine/strat_engine/thinpool/filesystem.rs

```diff
@@ -391,32 +391,22 @@ impl StratFilesystem {
 
     /// Find places where this filesystem is mounted.
     fn mount_points(&self) -> StratisResult<Vec<PathBuf>> {
-        // Use major:minor values to find mounts for this filesystem
-        let major = u64::from(self.thin_dev.device().major);
-        let minor = u64::from(self.thin_dev.device().minor);
-
-        let mut mount_data = String::new();
-        File::open("/proc/self/mountinfo")?.read_to_string(&mut mount_data)?;
-        let parser = libmount::mountinfo::Parser::new(mount_data.as_bytes());
-
-        let mut ret_vec = Vec::new();
-        for mp in parser {
-            match mp {
-                Ok(mount) =>
-                {
-                    #[allow(clippy::unnecessary_cast)]
-                    if mount.major as u64 == major && mount.minor as u64 == minor {
-                        ret_vec.push(PathBuf::from(&mount.mount_point));
-                    }
-                }
-                Err(e) => {
-                    let error_msg = format!("Error during parsing {:?}: {:?}", *self, e);
-                    return Err(StratisError::Msg(error_msg));
-                }
-            }
-        }
-
-        Ok(ret_vec)
+        let device = self.thin_dev.device();
+        let majmin = format!("{}:{}", device.major, device.minor);
+
+        procfs::process::Process::myself()
+            .and_then(|p| p.mountinfo())
+            .map(|infos| {
+                infos.into_iter()
+                    .filter(|mount| mount.majmin == majmin)
+                    .map(|mount| mount.mount_point)
```

Comment:
I think `filter_map` would be better here.

Reply:
Ok!